x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/formatted_field_udt.tsx

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable complexity */
```

Comment:
😅 ,  When I was working on this file, I had _many_ thoughts.

1. We should try to reduce the duplication between this and `formatted_field.tsx` since both will probably live together in the long run.
2. We should break the file up so we don't need this `disable-complexity` flag
3. It would be cool to collect telemetry on fields user's interact with the most to potentially set better defaults..not sure though

Anyways, updating/refactoring this and `formatted_fields.tsx` is it's own PR, and I'm happy to pick that work up, just let me know!

We should break this file up. Happy to help with this. We also should add tests for this. We can probably expand on the work of `formatted_field.test.tsx`.

Reply:
this was done here : a141c2b1c6ca7f9dce37bbed0e0001793c808482